src/services/trip.service.ts

```diff
@@ -0,0 +1,64 @@
+import { TripT } from "../validationSchemas/tripSchema";
+import { Trip } from "../models/trip.model";
+import { AppError } from "../utils/AppError";
+
+export const mongoCreateTrip = async (data: TripT): Promise<TripT> => {
+  try {
+    const trip = await Trip.create(data);
+    return trip;
+  } catch (error) {
+    throw new AppError(error.name, error.message, 500, "MongoDB");
+  }
+};
+
+export const mongoUpdateTrip = async ({
+  id,
+  data,
+}: {
+  id: string;
+  data: TripT;
+}): Promise<TripT> => {
+  try {
+    const updatedTrip = await Trip.findByIdAndUpdate(id, data, { new: true });
```

Comment:
what does the `new:true` mean?

Reply:
return the updated document after the update operation.